app/src/main/java/com/sopt/now/compose/container/UserRepository.kt

```diff
@@ -1,45 +1,61 @@
 package com.sopt.now.compose.container
 
 import android.content.SharedPreferences
+import android.util.Log
 import com.sopt.now.compose.models.User
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
 interface UserRepository {
     suspend fun getUserProfile(): User?
     suspend fun setUserProfile(user: User)
+
+    suspend fun getUserId(): String
+    suspend fun setUserId(userId: String)
 }
 
 class PreferenceUserRepository(
     val sharedPreferences: SharedPreferences
 ) : UserRepository {
-    override suspend fun getUserProfile(): User = withContext(Dispatchers.Main) {
+    override suspend fun getUserProfile(): User {
         sharedPreferences.run {
             val id = getString(ID_KEY, "") ?: ""
             val pw = getString(PW_KEY, "") ?: ""
             val nickname = getString(NICKNAME_KEY, "") ?: ""
             val mbti = getString(MBTI_KEY, "") ?: ""
-            return@withContext User(id, pw, nickname, mbti)
+            return User(id, pw, nickname, mbti)
         }
     }
 
     override suspend fun setUserProfile(user: User) {
         val edit = sharedPreferences.edit()
-        withContext(Dispatchers.Main) {
-            edit.run {
-                putString(ID_KEY, user.id)
-                putString(PW_KEY, user.pw)
-                putString(NICKNAME_KEY, user.nickName)
-                putString(MBTI_KEY, user.mbti)
-                commit()
-            }
+        edit.run {
+            putString(ID_KEY, user.id)
+            putString(PW_KEY, user.pw)
+            putString(NICKNAME_KEY, user.nickName)
+            putString(MBTI_KEY, user.phone)
+            commit()
         }
     }
 
+    override suspend fun getUserId(): String {
+        return sharedPreferences.getString(USER_ID_KEY, "").orEmpty()
+    }
```

Comment:
```suggestion
    override suspend fun getUserId() = sharedPreferences.getString(USER_ID_KEY, "").orEmpty()
```

코틀린 특성을 활용한다면 이렇게 짤 수도 있습니다. 다만 이는 개인 취향에 따라서 구현 방향이 달라지기에 취향껏 선택하시면 됩니다.